sdk/aqueduct/integrations/validation.py

```diff
@@ -0,0 +1,19 @@
+from typing import Any, Callable
+
+from aqueduct.utils.integration_validation import validate_integration_is_connected
+
+AnyFunc = Callable[..., Any]
+
+
+def validate_is_connected() -> Callable[[AnyFunc], AnyFunc]:
```

Comment:
I'd prefer exposing this as a method in integration base class and have all methods use `@self.validate_is_connected()`. In this way the method is tied to the integration and it's clear that we are ensuring 'this' integration is connected. It also avoids passing around `self` to an externally imported method (I'm not super sure if it's an antipattern but it feels weird to read :) )

Reply:
hmm I don't think I can use the `@self.validate_is_connected()` syntax.. My intellij errors and also https://stackoverflow.com/questions/11731136/class-method-decorator-with-self-arguments#:~:text=You%20can't.,use%20a%20different%20method%20entirely.

There's also a bit of a layering issue, where I'd have to move the `validate_integration_is_connected()` down to the models/ layer from utils/, which I feel doesn't make too much sense. This is all because the integration classes are defined in integrations/ and the base integration class is defined at a much lower level in `models`. I think these layering issues in the SDK should be addressed, but probably not in this PR. 

What do you think?

Reply:
yeah the current way looks fine then!